src/Features/Core/Portable/Formatting/FormattingFeatureOptions.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Options.Providers;
+
+namespace Microsoft.CodeAnalysis.Formatting
+{
+    internal static class FormattingFeatureOptions
```

Comment:
❔ Why here and not `FormattingOptions2`? The latter already contains `AutoFormattingOnReturn`.

I wouldn't complain if you went further and moved `AutoFormattingOnCloseBrace` and `FormatOnPaste` there at the same time. 😄

Reply:
Because this is what I've done for similar moves in the past, but I don't think they had a lower layer options bag. I can certainly move these to a different place instead.

Reply:
> AutoFormattingOnCloseBrace

This one is already in the Features layer, not the EditorFeatures layer, so I don't think that we need to move it.

> FormatOnPaste

This one is in the EditorFeatures layer, so I'll go ahead and move it so that we can eventually use it LSP without having to introduce a new EditorFeatures dependency.